chore: Add limitations for .NET agent Lambda function support. (#18183)
newrelic/docs-website: src/content/docs/apm/agents/net-agent/getting-started/net-agent-approaches-lambda.mdx

Since the agent automatically instruments most lambda functions, you can use the [agent NuGet package](https://www.nuget.org/packages/NewRelic.Agent#readme-body-tab) to monitor your lambda functions. You need to manually configure environment variables for your chosen deployment method (see our [installation guide](/install/dotnet/?deployment=nuget#nuget-linux)). This still requires that you set up either the [New Relic Lambda Extension or CloudWatch integration](/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/introduction-lambda/#how) to send your data to New Relic.

Limitations:

- Generic lambda Methods are not instrumented automatically. If your lambda method is a generic method, such as ``Task<TResponse> MyMethod<TRequest, TResponse>(TRequest, ILambdaContext)``, the .NET agent is currently not able to instrument that method.

> - AspNetCore lambda functions are currently not supported. AspNetCore lambda functions rely on a generic method that is registered as the lambda handler.

Reviewer: ```suggestion
- AspNetCore lambda functions are currently not supported. AspNetCore lambda functions rely on a generic method that's registered as the lambda handler.
```

- The [Lambda Annotations Framework](https://aws.amazon.com/blogs/developer/net-lambda-annotations-framework/) is currently not supported.

> - [ApiGatewayV2](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format) events are missing some context required for Distributed Tracing.

Reviewer: ```suggestion
- [ApiGatewayV2](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format) events are missing some context required for distributed tracing.
```

- Outbound Distributed Tracing for different AWS SDK calls (such as SQS) are not supported.

> - If your lambda function handler does not include an [ILambdaContext](https://docs.aws.amazon.com/lambda/latest/dg/csharp-context.html) parameter, the .NET agent will not be able to gather all of the expected information about your lambda function.

Reviewer: ```suggestion
- If your Lambda function handler does not include an [`ILambdaContext`](https://docs.aws.amazon.com/lambda/latest/dg/csharp-context.html) parameter, the .NET agent will not be able to gather all of the expected information about your lambda function.
```

> - .NET lambda functions build with the [Native AOT deployment method](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net7%2Cwindows) are not supported.

Reviewer: ```suggestion
- .NET Lambda functions build with the [Native AOT deployment method](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net7%2Cwindows) are not supported.
```

> - Generic lambda Methods are not instrumented automatically. If your lambda method is a generic method, such as ``Task<TResponse> MyMethod<TRequest, TResponse>(TRequest, ILambdaContext)``, the .NET agent is currently not able to instrument that method.
> - AspNetCore lambda functions are currently not supported. AspNetCore lambda functions rely on a generic method that's registered as the lambda handler.
> - The [Lambda Annotations Framework](https://aws.amazon.com/blogs/developer/net-lambda-annotations-framework/) is currently not supported.
> - [ApiGatewayV2](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format) events are missing some context required for distributed tracing.
> - Outbound Distributed Tracing for different AWS SDK calls (such as SQS) are not supported.
> - If your Lambda function handler does not include an [`ILambdaContext`](https://docs.aws.amazon.com/lambda/latest/dg/csharp-context.html) parameter, the .NET agent will not be able to gather all of the expected information about your lambda function.
> - .NET Lambda functions build with the [Native AOT deployment method](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net7%2Cwindows) are not supported.

Reviewer: ```suggestion
* Generic lambda Methods are not instrumented automatically. If your lambda method is a generic method, such as `Task<TResponse> MyMethod<TRequest, TResponse>(TRequest, ILambdaContext)`, the .NET agent is currently not able to instrument that method.
* AspNetCore lambda functions are currently not supported. AspNetCore lambda functions rely on a generic method that's registered as the lambda handler.
* The [Lambda Annotations Framework](https://aws.amazon.com/blogs/developer/net-lambda-annotations-framework/) is currently not supported.
* [ApiGatewayV2](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format) events are missing some context required for distributed tracing.
* Outbound distributed tracing for different AWS SDK calls (such as SQS) are not supported.
* If your Lambda function handler does not include an [`ILambdaContext`](https://docs.aws.amazon.com/lambda/latest/dg/csharp-context.html) parameter, the .NET agent will not be able to gather all of the expected information about your Lambda function.
* .NET Lambda functions built with the [Native AOT deployment method](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net7%2Cwindows) are not supported.
```
